src/viking.js

```diff
@@ -1,11 +1,46 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health , strength) {
+  this.health = health;
+  this.strength = strength;
+  }
+  attack(){
+  return this.strength;
+  }
+  receiveDamage(thedamage){
+  this.thedamage = thedamage;
+  this.health--;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+ constructor(name, health, strength){
+  super(health, strength);
+  this.name = name;
+ }
+ receiveDamage(thedamage) {
+  this.damage = thedamage;
+  if(this.health > 0) {
+  return ` ${this.name} has received ${thedamage} points of damage `;
+ } else {
+  return `${this.name} has died in the combat`;
+}
+ }
+ battleCry(){
+   return `Odin Owns You All`;
+ }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+  constructor(health,strength){
+    super(health, strength);
+    this.health = health;
+    this.strength = strength;
+  }
+ 
+}
```

Comment:
you can see the Saxon Class in the solutions, please let us know if you have any questions